diff -Naur onvif_srvd.ori/src/ServicePTZ.cpp onvif_srvd/src/ServicePTZ.cpp
--- onvif_srvd.ori/src/ServicePTZ.cpp	2020-07-18 18:15:02.000000000 +0200
+++ onvif_srvd/src/ServicePTZ.cpp	2020-08-04 14:47:03.000000000 +0200
@@ -8,6 +8,7 @@
 */
 
 
+#include <sstream>
 #include "soapPTZBindingService.h"
 #include "ServiceContext.h"
 #include "smacros.h"
@@ -29,10 +30,13 @@
 
 int GetPTZPreset(struct soap *soap, tt__PTZPreset* ptzp, int number)
 {
+    std::stringstream stream;
+
     ptzp->token = soap_new_std__string(soap);
-    *ptzp->token = std::to_string(number);
+    stream << number;
+    *ptzp->token = stream.str();
     ptzp->Name = soap_new_std__string(soap);
-    *ptzp->Name  = std::to_string(number);
+    *ptzp->Name  = stream.str();
     ptzp->PTZPosition = soap_new_tt__PTZVector(soap);;
     ptzp->PTZPosition->PanTilt          = soap_new_tt__Vector2D(soap);
     ptzp->PTZPosition->PanTilt->x       = 0.0;
